### PR TITLE
Fix build error with clang

### PR DESCRIPTION
**Pull request by tianjunwork**

### CODE REVIEW

**Comment by tianjunwork** [comment]
@joelsole could you review the change? Thanks.

**Comment by 1480c1** [comment]
@joelsole There is someone who did something like cpu_set_t in mac, <https://yyshen.github.io/2015/01/18/binding_threads_to_cores_osx.html>
Idk if it actually solves the cpu_set_t problem nor if it is all needed, but it does make the error go away. (At least for svt-hevc)

**Comment by tianjunwork** [comment]
@joelsole and @1480c1 the feature is low priority on MacOS. I will work on a patch to disable it on Mac.

**Comment by hassount** [comment]
@tianjunwork 
Any progress on the patch to get the MAC OS build running ?

**Comment by tianjunwork** [comment]
@1480c1 and @joelsole Travis CI showed passed. But I got error in MacOS build.

[ 95%] Linking C shared library ../../../../../../Bin/Release/libSvtAv1Enc.dylib
Undefined symbols for architecture x86_64:
  "_picture_average_kernel_sse2", referenced from:
      _avc_style_luma_interpolation_filter_pose_ssse3 in libASM_SSSE3.a(EbAvcStyleMcp_Intrinsic_SSSE3.c.o)
      _avc_style_luma_interpolation_filter_posf_ssse3 in libASM_SSSE3.a(EbAvcStyleMcp_Intrinsic_SSSE3.c.o)
      _avc_style_luma_interpolation_filter_posg_ssse3 in libASM_SSSE3.a(EbAvcStyleMcp_Intrinsic_SSSE3.c.o)
      _avc_style_luma_interpolation_filter_posi_ssse3 in libASM_SSSE3.a(EbAvcStyleMcp_Intrinsic_SSSE3.c.o)
      _avc_style_luma_interpolation_filter_posk_ssse3 in libASM_SSSE3.a(EbAvcStyleMcp_Intrinsic_SSSE3.c.o)
      _avc_style_luma_interpolation_filter_posp_ssse3 in libASM_SSSE3.a(EbAvcStyleMcp_Intrinsic_SSSE3.c.o)
      _avc_style_luma_interpolation_filter_posq_ssse3 in libASM_SSSE3.a(EbAvcStyleMcp_Intrinsic_SSSE3.c.o)
      ...
     (maybe you meant: _picture_average_kernel_sse2_intrin)
  "_picture_copy_kernel_sse2", referenced from:
      _avc_style_copy_sse2 in libASM_SSE2.a(EbAvcStyleMcp_Intrinsic_SSE2.c.o)
  "_zero_out_coeff16x16_sse2", referenced from:
      _pic_zero_out_coef_func_ptr_array in EbCodingLoop.c.o
  "_zero_out_coeff32x32_sse2", referenced from:
      _pic_zero_out_coef_func_ptr_array in EbCodingLoop.c.o
  "_zero_out_coeff4x4_sse", referenced from:
      _pic_zero_out_coef_func_ptr_array in EbCodingLoop.c.o
  "_zero_out_coeff8x8_sse2", referenced from:
      _pic_zero_out_coef_func_ptr_array in EbCodingLoop.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [../../../Bin/Release/libSvtAv1Enc.dylib] Error 1
make[2]: *** [Source/Lib/Codec/CMakeFiles/SvtAv1Enc.dir/all] Error 2
make[1]: *** [Source/App/CMakeFiles/SvtAv1EncApp.dir/rule] Error 2
make: *** [SvtAv1EncApp] Error 2

**Comment by 1480c1** [comment]
AppleClang prepends '_' to stuff, yasm doesn't

**Comment by tianjunwork** [comment]
> AppleClang prepends '_' to stuff, yasm doesn't

I don't have local MacOS to check the build. From travis result, is my patch fixed the build error introduced by PR#113?

**Comment by 1480c1** [comment]
Or was it the other way around?

**Comment by 1480c1** [comment]
Yes, now it doesn't complain that cpu_set_t isn't defined

**Comment by 1480c1** [comment]
Now we are back to pre #113  error
<https://travis-ci.org/OpenVisualCloud/SVT-AV1/jobs/499582550> From 112

**Comment by tianjunwork** [comment]
> Yes, now it doesn't complain that cpu_set_t isn't defined

Thanks!

**Comment by 1480c1** [comment]
how come the patch is from 3 hours in the future? lol.

**Comment by tianjunwork** [comment]
> how come the patch is from 3 hours in the future? lol.

The future me is still working? Diner time.

**Comment by joelsole** [comment]
This seems to be related to PR108:

https://travis-ci.org/OpenVisualCloud/SVT-AV1/jobs/499506565

**Comment by joelsole** [comment]
Related to PR #108: 
https://travis-ci.org/OpenVisualCloud/SVT-AV1/jobs/499506565

@spawlows , can you have a look?

**Comment by 1480c1** [comment]
```
nm Source/Lib/ASM_SSE2/CMakeFiles/ASM_SSE2.dir/EbPictureOperators_SSE2.asm.o
0000000000000467 t Label_PictureAverageKernel_SSE2_WIDTH12
000000000000074d t Label_PictureAverageKernel_SSE2_WIDTH16
00000000000006f4 t Label_PictureAverageKernel_SSE2_WIDTH24
000000000000068f t Label_PictureAverageKernel_SSE2_WIDTH32
00000000000004e0 t Label_PictureAverageKernel_SSE2_WIDTH4
00000000000005fb t Label_PictureAverageKernel_SSE2_WIDTH48
000000000000052c t Label_PictureAverageKernel_SSE2_WIDTH64
00000000000004b2 t Label_PictureAverageKernel_SSE2_WIDTH8
000000000000050e t Label_PictureAverageKernel_SSE2_WIDTH_Big
000000000000001c t Label_PictureCopyKernel_SSE2_WIDTH12
00000000000000b9 t Label_PictureCopyKernel_SSE2_WIDTH128
0000000000000234 t Label_PictureCopyKernel_SSE2_WIDTH16
00000000000001fd t Label_PictureCopyKernel_SSE2_WIDTH24
00000000000001c6 t Label_PictureCopyKernel_SSE2_WIDTH32
0000000000000074 t Label_PictureCopyKernel_SSE2_WIDTH4
0000000000000179 t Label_PictureCopyKernel_SSE2_WIDTH48
0000000000000116 t Label_PictureCopyKernel_SSE2_WIDTH64
0000000000000053 t Label_PictureCopyKernel_SSE2_WIDTH8
0000000000000095 t Label_PictureCopyKernel_SSE2_WIDTH_Big
000000000000037d t Label_ZeroOutCoeff32x32_SSE2_01
0000000000000787 T Log2f_SSE2
0000000000000787 T _Log2f_SSE2
0000000000000440 T _PictureAverageKernel_SSE2
0000000000000000 T _PictureCopyKernel_SSE2
00000000000002a9 T _ZeroOutCoeff16x16_SSE2
000000000000036a T _ZeroOutCoeff32x32_SSE2
0000000000000255 T _ZeroOutCoeff4x4_SSE
0000000000000272 T _ZeroOutCoeff8x8_SSE2
0000000000000440 T picture_average_kernel_sse2
0000000000000000 T picture_copy_kernel_sse2
00000000000002a9 T zero_out_coeff16x16_sse2
000000000000036a T zero_out_coeff32x32_sse2
0000000000000255 T zero_out_coeff4x4_sse
0000000000000272 T zero_out_coeff8x8_sse2
```

```

0000000000001610 S _AvcStyleLumaIFCoeff8_SSSE3
00000000000000a0 T _avc_style_luma_interpolation_filter_horizontal_ssse3_intrin
0000000000000000 T _avc_style_luma_interpolation_filter_pose_ssse3
0000000000000860 T _avc_style_luma_interpolation_filter_posf_ssse3
0000000000000a80 T _avc_style_luma_interpolation_filter_posg_ssse3
0000000000000b30 T _avc_style_luma_interpolation_filter_posi_ssse3
0000000000000d60 T _avc_style_luma_interpolation_filter_posj_ssse3
0000000000000f50 T _avc_style_luma_interpolation_filter_posk_ssse3
0000000000001190 T _avc_style_luma_interpolation_filter_posp_ssse3
0000000000001240 T _avc_style_luma_interpolation_filter_posq_ssse3
0000000000001450 T _avc_style_luma_interpolation_filter_posr_ssse3
00000000000003d0 T _avc_style_luma_interpolation_filter_vertical_ssse3_intrin
                 U _picture_average_kernel_sse2

**Comment by 1480c1** [comment]
The above are the dumps from the asm and I guess their respective c obj which tries to refer to them.

**Comment by 1480c1** [comment]
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ac8b7a..ea3955a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,7 @@ set(CMAKE_ASM_NASM_SOURCE_FILE_EXTENSIONS asm)

 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     set(CMAKE_ASM_NASM_OBJECT_FORMAT macho64)
-    set(CMAKE_ASM_NASM_FLAGS "-DUNIX64")
+    set(CMAKE_ASM_NASM_FLAGS "-DUNIX64 -DPREFIX")
     set(CMAKE_MACOSX_RPATH 1)
 endif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")

diff --git a/Source/Lib/ASM_SSE2/EbPictureOperators_SSE2.asm b/Source/Lib/ASM_SSE2/EbPictureOperators_SSE2.asm
index c4399fd..5e4ce0c 100644
--- a/Source/Lib/ASM_SSE2/EbPictureOperators_SSE2.asm
+++ b/Source/Lib/ASM_SSE2/EbPictureOperators_SSE2.asm
@@ -626,7 +626,7 @@ Label_PictureAverageKernel_SSE2_WIDTH16:
     ret

 ; ----------------------------------------------------------------------------------------
-    cglobal _Log2f_SSE2
+;    cglobal _Log2f_SSE2
     cglobal Log2f_SSE2
     bsr rax, r0
     ret
diff --git a/Source/Lib/ASM_SSE2/x64RegisterUtil.asm b/Source/Lib/ASM_SSE2/x64RegisterUtil.asm
index 151a307..d1bf24d 100644
--- a/Source/Lib/ASM_SSE2/x64RegisterUtil.asm
+++ b/Source/Lib/ASM_SSE2/x64RegisterUtil.asm
@@ -21,7 +21,7 @@ section .text
 ; So if RunEmms() is called according to the above cases,
 ; then the "emms" instruction in all other assembly functions can be removed.

-cglobal _RunEmms
+;cglobal _RunEmms
 cglobal RunEmms
     emms
     ret
```
Will allow it to build

**Comment by 1480c1** [comment]
Although, actually running it maybe fails? (Maybe 2011 macbook pro is just too bad?)
```
(lldb) settings set -- target.run-args  "-n" "120" "-w" "1920" "-h" "1080" "-fps-num" "2997" "-fps-denom" "50" "-bit-depth" "8" "-i" "../Release/Morocco1080.yuv" "-b" "Morroco.ivf" "-enc-mode" "4"
(lldb) run
Process 17357 launched: './SvtAv1EncApp' (x86_64)
-------------------------------------------
SVT-AV1 Encoder
SVT [version]:  SVT-AV1 Encoder Lib v0.4.0
SVT [build]  :  GCC 4.2.1        64 bit
LIB Build date: Mar  6 2019 19:01:52
-------------------------------------------
Number of logical cores available: 8
Number of PPCS 73
-------------------------------------------
SVT [config]: Main Profile      Tier (auto)     Level (auto)
SVT [config]: EncoderMode                                                       : 4
SVT [config]: EncoderBitDepth / CompressedTenBitFormat                          : 8 / 0
SVT [config]: SourceWidth / SourceHeight                                        : 1920 / 1080
SVT [config]: Fps_Numerator / Fps_Denominator / Gop Size / IntraRefreshType     : 2997 / 50 / 64 / 1
SVT [config]: HierarchicalLevels / BaseLayerSwitchMode / PredStructure          : 4 / 0 / 2
SVT [config]: BRC Mode / QP  / LookaheadDistance / SceneChange                  : CQP / 50 / 33 / 0
-------------------------------------------
Encoding          Process 17384 stopped
* thread #2, stop reason = EXC_BAD_INSTRUCTION (code=EXC_I386_INVOP, subcode=0x0)
    frame #0: 0x0000000100419ae9 libSvtAv1Enc.dylib`Compute4xMSadSub_AVX2_INTRIN(src="[aBunchOfCharactersThatLookLikeAnEscCode*90]", refStride=64, height=4, width=4) at EbComputeSAD_Intrinsic_AVX2.c:3364
   3361     __m256i mm256_ref;
   3362
   3363     for (y = 0; y < hsteps; y++) {
-> 3364         mm256_src = _mm256_setr_epi32(*(uint32_t*)src,
   3365             *(uint32_t*)(src + 1 * src_stride),
   3366             *(uint32_t*)(src + 2 * src_stride),
   3367             *(uint32_t*)(src + 3 * src_stride),
Target 0: (SvtAv1EncApp) stopped.
```

**Comment by spawlows** [comment]
@joelsole I moved back onc commit on my branch, and i still have that error. 
https://travis-ci.com/spawlows/SVT-AV1/jobs/182937274

I will try to tests with revert all commits and try to find reason.

**Comment by 1480c1** [comment]
```
From db5c323a0d4d2242938e27f0227f39468c1bdcc2 Mon Sep 17 00:00:00 2001
From: Chris Degawa <ccom@randomderp.com>
Date: Wed, 6 Mar 2019 19:53:20 -0600
Subject: [PATCH] Add '_' prefix for asm functions if macOS

Due to AppleClang prefixing C functions with '_', when linking with
yasm produced files, there is no junction between the two functions
causeing symbols not found for x86_64 with macOS builds
---
 CMakeLists.txt                                  |  8 ++++----
 Source/Lib/ASM_SSE2/CMakeLists.txt              | 12 ++++--------
 Source/Lib/ASM_SSE2/EbPictureOperators_SSE2.asm | 11 +++++++----
 Source/Lib/ASM_SSE2/x64RegisterUtil.asm         |  9 ++++++---
 Source/Lib/ASM_SSE4_1/CMakeLists.txt            | 12 ++++--------
 Source/Lib/ASM_SSSE3/CMakeLists.txt             |  9 ++++-----
 Source/Lib/Codec/CMakeLists.txt                 | 16 ++++++++++------
 7 files changed, 39 insertions(+), 38 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ac8b7a..f6a877d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
-# 
+#
 # Copyright(c) 2018 Intel Corporation
 # SPDX - License - Identifier: BSD - 2 - Clause - Patent
-# 
+#
 
 cmake_minimum_required(VERSION 3.5)
 set(project_name "svt-av1")
@@ -21,12 +21,12 @@ set(CMAKE_ASM_NASM_SOURCE_FILE_EXTENSIONS asm)
 
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     set(CMAKE_ASM_NASM_OBJECT_FORMAT macho64)
-    set(CMAKE_ASM_NASM_FLAGS "-DUNIX64")
+    set(CMAKE_ASM_NASM_FLAGS "-DUNIX64 -DPREFIX")
     set(CMAKE_MACOSX_RPATH 1)
 endif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 
 if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    set(CMAKE_C_FLAGS "-Wall -Wextra -Wformat -Wformat-security -mavx -fstack-protector-strong -fPIE -fPIC -D_FORTIFY_SOURCE=2 ")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wformat -Wformat-security -mavx -fstack-protector-strong -fPIE -fPIC -D_FORTIFY_SOURCE=2")
     set(CMAKE_C_FLAGS_RELEASE "-O2")
     set(CMAKE_C_FLAGS_DEBUG "-O0 -g")
 	SET(CMAKE_EXE_LINKER_FLAGS "-z noexecstack -z relro -z now -pie ")
diff --git a/Source/Lib/ASM_SSE2/CMakeLists.txt b/Source/Lib/ASM_SSE2/CMakeLists.txt
index bcf8e50..7592748 100644
--- a/Source/Lib/ASM_SSE2/CMakeLists.txt
+++ b/Source/Lib/ASM_SSE2/CMakeLists.txt
@@ -1,7 +1,7 @@
-# 
+#
 # Copyright(c) 2019 Intel Corporation
 # SPDX - License - Identifier: BSD - 2 - Clause - Patent
-# 
+#
 
 # ASM_SSE2 Directory CMakeLists.txt
 
@@ -12,10 +12,10 @@ include_directories (${PROJECT_SOURCE_DIR}/Source/Lib/C_DEFAULT/)
 include_directories (${PROJECT_SOURCE_DIR}/Source/Lib/ASM_SSE2/)
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-    SET(CMAKE_C_FLAGS "/MP")    
+    SET(CMAKE_C_FLAGS "/MP")
 endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
 
-if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2")
 endif()
 
@@ -31,7 +31,3 @@ file(GLOB all_files
 add_library(ASM_SSE2
     ${all_files}
 )
-
-
-
-
diff --git a/Source/Lib/ASM_SSE2/EbPictureOperators_SSE2.asm b/Source/Lib/ASM_SSE2/EbPictureOperators_SSE2.asm
index c4399fd..69ba0b4 100644
--- a/Source/Lib/ASM_SSE2/EbPictureOperators_SSE2.asm
+++ b/Source/Lib/ASM_SSE2/EbPictureOperators_SSE2.asm
@@ -1,7 +1,7 @@
-; 
+;
 ; Copyright(c) 2019 Intel Corporation
 ; SPDX - License - Identifier: BSD - 2 - Clause - Patent
-; 
+;
 
 %include "x64inc.asm"
 %include "x64Macro.asm"
@@ -526,7 +526,7 @@ Label_PictureAverageKernel_SSE2_WIDTH48:
     pavgb           xmm0,            xmm6
     movdqu          xmm6,            [src1+16]
     pavgb           xmm1,            xmm6
-    movdqu          xmm6,            [src1+32]            
+    movdqu          xmm6,            [src1+32]
     pavgb           xmm2,            xmm6
     movdqu          xmm6,            [src1+src1_stride]
     pavgb           xmm3,            xmm6
@@ -626,8 +626,11 @@ Label_PictureAverageKernel_SSE2_WIDTH16:
     ret
 
 ; ----------------------------------------------------------------------------------------
+%ifidn __OUTPUT_FORMAT__, macho64
+    cglobal Log2f_SSE2
+%else
     cglobal _Log2f_SSE2
     cglobal Log2f_SSE2
+%endif
     bsr rax, r0
     ret
-
diff --git a/Source/Lib/ASM_SSE2/x64RegisterUtil.asm b/Source/Lib/ASM_SSE2/x64RegisterUtil.asm
index 151a307..7bbe92f 100644
--- a/Source/Lib/ASM_SSE2/x64RegisterUtil.asm
+++ b/Source/Lib/ASM_SSE2/x64RegisterUtil.asm
@@ -1,7 +1,7 @@
-; 
+;
 ; Copyright(c) 2019 Intel Corporation
 ; SPDX - License - Identifier: BSD - 2 - Clause - Patent
-; 
+;
 
 %include "x64inc.asm"
 
@@ -20,9 +20,12 @@ section .text
 
 ; So if RunEmms() is called according to the above cases,
 ; then the "emms" instruction in all other assembly functions can be removed.
-
+%ifidn __OUTPUT_FORMAT__, macho64
+cglobal RunEmms
+%else
 cglobal _RunEmms
 cglobal RunEmms
+%endif
     emms
     ret
 
diff --git a/Source/Lib/ASM_SSE4_1/CMakeLists.txt b/Source/Lib/ASM_SSE4_1/CMakeLists.txt
index 11c4d8f..f815b5b 100644
--- a/Source/Lib/ASM_SSE4_1/CMakeLists.txt
+++ b/Source/Lib/ASM_SSE4_1/CMakeLists.txt
@@ -1,7 +1,7 @@
-# 
+#
 # Copyright(c) 2019 Intel Corporation
 # SPDX - License - Identifier: BSD - 2 - Clause - Patent
-# 
+#
 
 # ASM_SSE4.1 Directory CMakeLists.txt
 
@@ -15,10 +15,10 @@ include_directories (${PROJECT_SOURCE_DIR}/Source/Lib/ASM_SSE4_1/)
 include_directories (${PROJECT_SOURCE_DIR}/Source/Lib/ASM_AVX2/)
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-    SET(CMAKE_C_FLAGS "/MP")    
+    SET(CMAKE_C_FLAGS "/MP")
 endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
 
-if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse4.1")
 endif()
 
@@ -35,7 +35,3 @@ file(GLOB all_files
 add_library(ASM_SSE4_1
     ${all_files}
 )
-
-
-
-
diff --git a/Source/Lib/ASM_SSSE3/CMakeLists.txt b/Source/Lib/ASM_SSSE3/CMakeLists.txt
index 1658aa9..97e7265 100644
--- a/Source/Lib/ASM_SSSE3/CMakeLists.txt
+++ b/Source/Lib/ASM_SSSE3/CMakeLists.txt
@@ -1,7 +1,7 @@
-ï»¿# 
+ï»¿#
 # Copyright(c) 2019 Intel Corporation
 # SPDX - License - Identifier: BSD - 2 - Clause - Patent
-# 
+#
 
 # ASM_SSSE3 Directory CMakeLists.txt
 
@@ -15,10 +15,10 @@ include_directories (${PROJECT_SOURCE_DIR}/Source/Lib/ASM_SSE4_1/)
 include_directories (${PROJECT_SOURCE_DIR}/Source/Lib/ASM_AVX2/)
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-    SET(CMAKE_C_FLAGS "/MP")    
+    SET(CMAKE_C_FLAGS "/MP")
 endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
 
-if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mssse3")
 endif()
 
@@ -34,4 +34,3 @@ file(GLOB all_files
 add_library(ASM_SSSE3
     ${all_files}
 )
-
diff --git a/Source/Lib/Codec/CMakeLists.txt b/Source/Lib/Codec/CMakeLists.txt
index 8d48d0a..ac8fd86 100644
--- a/Source/Lib/Codec/CMakeLists.txt
+++ b/Source/Lib/Codec/CMakeLists.txt
@@ -1,7 +1,7 @@
-# 
+#
 # Copyright(c) 2019 Intel Corporation
 # SPDX - License - Identifier: BSD - 2 - Clause - Patent
-# 
+#
 # EncoderLib Directory CMakeLists.txt
 
 # Include Encoder Subdirectories
@@ -24,18 +24,22 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/Bin/${CMAKE_BUILD_TYPE}
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/Bin/${CMAKE_BUILD_TYPE}/)
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-   SET(CMAKE_C_FLAGS "/MP")    
+   SET(CMAKE_C_FLAGS "/MP")
 endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+if(UNIX)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mavx")
+endif()
+
 
 file(GLOB all_files
     "*.h"
     "*.c")
- 
+
 # Encoder Lib Source Files
 add_library(SvtAv1Enc SHARED
     ${all_files}
 )
-    
+
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
 target_link_libraries(SvtAv1Enc
     C_DEFAULT
@@ -60,7 +64,7 @@ endif()
 if(NOT DEFINED CMAKE_INSTALL_INCLUDEDIR)
     set(CMAKE_INSTALL_INCLUDEDIR "include/svt-av1")
 endif()
-    
+
 configure_file(../pkg-config.pc.in ${CMAKE_BINARY_DIR}/SvtAv1Enc.pc @ONLY)
 install(FILES ${CMAKE_BINARY_DIR}/SvtAv1Enc.pc DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 install(TARGETS SvtAv1Enc DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-- 
2.20.1.windows.1

```

To be applied on top of  90d1aaf619cb0799be416ef31e1f895dd4087082.
<https://travis-ci.com/1480c1/SVT-AV1/builds/103468757>
Allows for proper building of MacOS and linux.
Due to the inability to chose to not use AVX2, I am unable to completely test if this works since when I try to run SvtAv1EncApp, it attempts to use AVX2 instructions when AVX2 is not supported on my machine, resulting in EXC_BAD_INSTRUCTION.

I will later try to see if I can manually build without avx2 and see if it will run.

**Comment by spawlows** [comment]
I found commit after which stops build macOS: c27a187d2e83749962d35e2b0a6b74f385f8b58c
Previous commit on macOS  have warnings, but build end success.
https://travis-ci.com/spawlows/SVT-AV1/jobs/182955298

**Comment by spawlows** [comment]
@tianjunwork Please add my commit fa4a8b14822014c7c51f19b6b6ccf7f217b89ca9 to fix. 
It resolve issue with:
Undefined symbols for architecture x86_64:
_picture_average_kernel_sse2
_picture_copy_kernel_sse2
_zero_out_coeff16x16_sse2
_zero_out_coeff32x32_sse2
_zero_out_coeff4x4_sse
_zero_out_coeff8x8_sse2

**Comment by 1480c1** [comment]
```
Process 27225 launched: './SvtAv1EncApp' (x86_64)
-------------------------------------------
SVT-AV1 Encoder
SVT [version]:  SVT-AV1 Encoder Lib v0.4.0
SVT [build]  :  GCC 4.2.1        64 bit
LIB Build date: Mar  7 2019 07:04:40
-------------------------------------------
Number of logical cores available: 8
Number of PPCS 73
-------------------------------------------
SVT [config]: Main Profile      Tier (auto)     Level (auto)
SVT [config]: EncoderMode                                                      :                                                                                                                                                              7
SVT [config]: EncoderBitDepth / CompressedTenBitFormat                         :                                                                                                                                                              8 / 0
SVT [config]: SourceWidth / SourceHeight                                       :                                                                                                                                                              1920 / 1080
SVT [config]: Fps_Numerator / Fps_Denominator / Gop Size / IntraRefreshType    :                                                                                                                                                              2997 / 50 / 64 / 1
SVT [config]: HierarchicalLevels / BaseLayerSwitchMode / PredStructure         :                                                                                                                                                              4 / 0 / 2
SVT [config]: BRC Mode / QP  / LookaheadDistance / SceneChange                 :                                                                                                                                                              CQP / 50 / 33 / 0
-------------------------------------------
Encoding       120
SUMMARY --------------------------------- Channel 1  --------------------------------
Total Frames            Frame Rate              Byte Count              Bitrate
         120            59.94 fps                    95502              381.63 kbps


Channel 1
Average Speed:          0.364 fps
Total Encoding Time:    329861 ms
Total Execution Time:   331491 ms
Average Latency:        224476 ms
Max Latency:            308099 ms
Encoder finished
Process 27225 exited with status = 0 (0x00000000)
```

Need to wait a bit to actually check if it is correct.

**Comment by 1480c1** [comment]
image

**Comment by 1480c1** [comment]
Looks fine to me

**Comment by tianjunwork** [comment]
Thanks @1480c1 and @spawlows, @hassount this PR is ready to merge. It fixed multiple build failures on MacOS and with clang.

File: Source/Lib/Codec/EbEncHandle.c [hunk 1]
```diff
@@ -462,11 +464,13 @@ EbErrorType LoadDefaultBufferConfigurationSettings(
 
     unsigned int lpCount = GetNumProcessors();
     unsigned int coreCount = lpCount;
+#if !defined(__APPLE__)
```
[the remark this answers is not in the document]
Comment (tianjunwork) [reply Source/Lib/Codec/EbEncHandle.c]: agree. I originally thought with this mark it's easier for developer to read code and add the support later for __APPLE__.

**Comment by tianjunwork** [comment]
@1480c1 My vlc shows error "VLC could not decode the format "av01" (AOMedia's AV1 Video". Is your vlc a special build? 
By the way, my input file is webm/mkv container.

**Comment by 1480c1** [comment]
> @1480c1 My vlc shows error "VLC could not decode the format "av01" (AOMedia's AV1 Video". Is your vlc a special build?
> By the way, my input file is webm/mkv container.

@tianjunwork mine is 3.0.6 (jenkins@728791c5f812 jan 9 2019). I did use windows for playback because I don't trust the mac to be able to decode well.
I used yuv (enc)-> ivf. I did not know if you could use other containers as input, or do you mean input for vlc? yuv -> webm/mkv?

**Comment by 1480c1** [comment]
Installed through chocolatey
image


image

**Comment by tianjunwork** [comment]
@1480c1 Thank you very much. I will check my vlc version. By the way, I output av1 bitstream with mkv container.

**Comment by 1480c1** [comment]
I will try yuv direct mux into mkv later

**Comment by 1480c1** [comment]
image

image
mkv seems to work for me

**Comment by tianjunwork** [comment]
Hi @jbeich, could you review [https://github.com/OpenVisualCloud/SVT-AV1/pull/130/commits/a9d64ba96a8a1b1fd637e69116268ceaa4282a47](url). And get it merged?

**Comment by 1480c1** [comment]
After this gets merged, can the same fixes be applied to the rest of the encoders?

**Comment by kylophone** [comment]
Merged, thanks!

**Pull request merged**